integration/kubernetes/k8s-number-cpus.bats

```diff
@@ -31,9 +31,11 @@ setup() {
 		# Get number of cpus
 		number_cpus=$(kubectl exec pod/"$pod_name" -c "$container_name" \
 			-- sh -c "$num_cpus_cmd")
-		# Verify number of cpus
-		[ "$number_cpus" -le "$max_number_cpus" ]
-		[ "$number_cpus" -eq "$max_number_cpus" ] && break
+		if [[ "$number_cpus" =~ ^[0-9]+$ ]]; then
```

Comment:
@wainersm Why is this empty sometimes?

Reply:
Hi @amshinde , that's actually a good question and I don't know. Worth a separate issue?

The test has a pod with c1 and c2 containers. The former is always spinning but the later dies every 10 seconds, thus c2 is restarted by k8s from time to time. My first thought was that maybe by the time it read from /proc/cpuinfo on c1, the runtime is hot-unplugging the CPU from (dead) c2 thus the proc file is still in undetermined state. I really don't know if such as hypothesis makes sense...

@c3d any clue?

Reply:
@gkurz do you know what may be happening?

Reply:
Weird -- why not just sleep infinity in that workload?

Also -- related -- there seems to be a LOT of overlap between this test and k8s-cpu-ns.bats -- perhaps we should consolidate these two tests?

Reply:
Also, it may be useful to augment the debug information available as part of the teardown. Either printing number of CPUs, or re-running the kubectl exec exec "$pod_name" -c "$container_name" -- sh -c "$num_cpus_cmd" ?